lib/core/auth/bloc/auth_bloc.dart

```diff
@@ -184,8 +188,229 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
       }
     });
 
+    /// Log in with OAuth Provider to get a code.
+    on<OAuthLoginAttempt>((event, emit) async {
+      LemmyClient lemmyClient = LemmyClient.instance;
+      String originalBaseUrl = lemmyClient.lemmyApiV3.host;
+      String instance = event.instance ?? state.oauthInstance!;
+      ProviderView provider = event.provider ?? state.oauthProvider!;
+
+      try {
+        emit(state.copyWith(status: AuthStatus.loading, account: null, isLoggedIn: false));
+
+        if (instance.startsWith('https://')) instance = instance.replaceAll('https://', '');
+        lemmyClient.changeBaseUrl(instance);
+
+        // Build oauth provider url.
+        var authorizationEndpoint = Uri.parse(provider.authorizationEndpoint);
+        String redirectUri = "https://thunderapp.dev/oauth/callback"; // This must end in /oauth/callback.
```

Comment:
Do you want to maybe pull this URL out into the globals file or at least define it as a constant in this file? That also lets you reuse it in part 2.

Reply:
I made it a constant at the top of this file.